src/main/java/gift/repository/WishRepository.java

```diff
@@ -1,16 +1,24 @@
 package gift.repository;
 
 import gift.domain.Wish;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Modifying;
+import org.springframework.data.jpa.repository.Query;
 
 import java.util.List;
 
-public interface WishRepository {
+public interface WishRepository extends JpaRepository<Wish, Long> {
+    List<Wish> findAllByMemberId(Long memberId);
 
-    void save(Long memberId, Long productId, int quantity);
+    @Query("SELECT COUNT(w) > 0 FROM Wish w WHERE w.member.id = :memberId AND w.product.id = :productId")
+    boolean existsByMemberIdAndProductId(Long memberId, Long productId);
```

Comment:
```suggestion
    boolean existsByMemberIdAndProductId(Long memberId, Long productId);
```
이 경우 메서드 이름 기반 쿼리를 사용하면 성능상의 이점이 있습니다. 왜일까요?

Reply:
피드백 주신 내용을 바탕으로 학습해보았습니다! 메서드 이름 기반의 existsBy를 사용하면 JPA가 내부적으로 EXISTS 쿼리나 LIMIT 1과 같은 최적화된 쿼리를 생성해줘서 성능상 이점이 있는 반면에 @Query로 COUNT 쿼리를 작성하면 데이터를 끝까지 스캔해서 개수를 세고 0보다 큰지 비교하기 때문에, 데이터가 많아질수록 비효율적입니다. 그래서 단순 존재 여부를 판단할 때는 메서드 이름 기반의 existsBy를 사용해야 성능상의 이점이 있습니다...